src/Avalonia.Controls/WrapPanel.cs

```diff
@@ -95,40 +127,51 @@ IInputElement INavigableContainer.GetControl(NavigationDirection direction, IInp
         /// <inheritdoc/>
         protected override Size MeasureOverride(Size constraint)
         {
-            var curLineSize = new UVSize(Orientation);
-            var panelSize = new UVSize(Orientation);
-            var uvConstraint = new UVSize(Orientation, constraint.Width, constraint.Height);
-
-            var childConstraint = new Size(constraint.Width, constraint.Height);
-
-            for (int i = 0, count = Children.Count; i < count; i++)
+            double itemWidth = ItemWidth;
+            double itemHeight = ItemHeight;
+            var orientation = Orientation;
+            var children = Children;
+            var curLineSize = new UVSize(orientation);
+            var panelSize = new UVSize(orientation);
+            var uvConstraint = new UVSize(orientation, constraint.Width, constraint.Height);
+            bool itemWidthSet = !double.IsNaN(itemWidth);
+            bool itemHeightSet = !double.IsNaN(itemHeight);
+
+            var childConstraint = new Size(
+                itemWidthSet ? itemWidth : constraint.Width,
+                itemHeightSet ? itemHeight : constraint.Height);
+            
+            for (int i = 0, count = children.Count; i < count; i++)
             {
-                var child = Children[i];
-                if (child == null) continue;
-
-                //Flow passes its own constrint to children
-                child.Measure(childConstraint);
-
-                //this is the size of the child in UV space
-                var sz = new UVSize(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
-
-                if (MathUtilities.GreaterThan(curLineSize.U + sz.U, uvConstraint.U)) //need to switch to another line
+                var child = children[i];
+                if (child != null)
                 {
-                    panelSize.U = Max(curLineSize.U, panelSize.U);
-                    panelSize.V += curLineSize.V;
-                    curLineSize = sz;
+                    //Flow passes its own constrint to children
```

Comment:
Could you change all `constrint` to `constraint` in this file?

Reply:
Resolved